client/lib/css-hot-reload/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+var io = require( 'socket.io-client' );
+
+/**
+ * Internal dependencies
+ */
+import { cssBuildFailed, cssBuilding } from 'state/application/actions';
+import i18n from 'lib/mixins/i18n';
+
+/**
+ * @returns {boolean} 
+ */
+function isChanged( href, changedFiles ) {
+	// "/calypso/style-debug.css?v=5a1db7fee7" -> "/calypso/style-debug.css"
+	var css = href.split('?')[0];
+	var i, re;
+		
+	for( i = 0; i < changedFiles.length; ++i ) {
+		re = new RegExp( '/' + changedFiles[i] + '$' );
+		if ( css.match( re ) ) {
+			return true;
+		}
+	}
+	
+	return false;
+}
+
+var CssHotReload = {
+
+	/**
+	 * Initialize client CSS hot-reload handler
+	 */
+	init: function( reduxStore ) {
+
+		var namespace = location.protocol + '//' + location.host + '/css-hot-reload';
+		var socket = io.connect( namespace );
+
+		socket.on( 'css-hot-reload', function( data ) {
+			switch( data.status ) {
+				case 'reload':
+					// Turn HTMLCollection to standard list
+					var elems = document.head.getElementsByTagName('LINK');
+					elems = [].slice.call( elems );
+					elems.forEach( function( oldLink ) { 
+						if ( ( 'href' in oldLink ) && isChanged( oldLink.href, data.changedFiles ) ) {
+							console.log( 'Reloading CSS: ', oldLink );
+							// Remove old .css and insert new one in the same spot
+							var newLink = document.createElement( 'LINK' );
```

Comment:
This function will automatically convert its argument to lowercase. Is there a reason you have it using uppercase here and in line 44 above?